src/Components/Web/src/Forms/InputBase.cs

```diff
@@ -159,7 +155,7 @@ protected InputBase()
         /// <param name="result">An instance of <typeparamref name="TValue"/>.</param>
         /// <param name="validationErrorMessage">If the value could not be parsed, provides a validation error message.</param>
         /// <returns>True if the value could be parsed; otherwise false.</returns>
-        protected abstract bool TryParseValueFromString(string? value, [MaybeNull] out TValue result, [NotNullWhen(false)] out string? validationErrorMessage);
+        protected abstract bool TryParseValueFromString(string? value, out TValue? result, [NotNullWhen(false)] out string? validationErrorMessage);
```

Comment:
This is an interesting edge case for me. I believe this signature is closest to the intended semantics:

```suggestion
        protected abstract bool TryParseValueFromString(string? value, [MaybeNullWhen(false)] out TValue result, [NotNullWhen(false)] out string? validationErrorMessage);
```

Reply:
>  If TValue is a non-nullable reference type, it does make a claim that result is non-null when the method returns true.

It might be problematic in cases like this: https://github.com/dotnet/aspnetcore/blob/c42d1a3858906cd372eb787697f00c89ef2cde55/src/Components/Web/src/Forms/InputText.cs#L35-L40. The true-ness here indicates if we were able to parse this value if it's a string representation of a .NET type: https://github.com/dotnet/aspnetcore/blob/5bc2c49ed560663c57198df3c6e92273c319173b/src/Components/Web/src/Forms/InputDate.cs#L51-L81

Reply:
Given this, do you still think this would be right annotation to apply?

Reply:
`InputText` extends `InputBase<string?>`, so this falls into the first category I mentioned (i.e. it will still be `out string?` even with my proposed change). The `InputDate` case is also not a problem, though you will need suppressions or minor rewrite because the compiler doesn't understand how to follow the indirection through the local variable `bool success`.